### ISSUE TITLE
@Assert\NotNull annotation always added for attributes with custom name

### ISSUE THREAD

**Issue by JulienCheny** [api-platform/schema-generator#134]
On the last api-platform distribution.

### Actual Result
When I generate entity with an attribute that is not in schema.org, `@Assert\NotNull` annotation is always added.

### Expected Result
`@Assert\NotNull` should be added only when I write `nullable: false`.

#### /api-platform/api/config/schema.yml
```
types:
    Thing:
        parent: false 
        properties:
            name: ~
            customName: {range: Text}
            otherCustomName: {range: Text, nullable: true, ormColumn: 'type="text", nullable=true'}
```

#### /api-platform/api/src/Entity/Thing.php
```
class Thing
{
[...]
    /**
     * @var string|null the name of the item
     *
     * @ORM\Column(type="text", nullable=true)
     * @ApiProperty(iri="http://schema.org/name")
     */
    private $name;

    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $customName;

    /**
     * @var string
     *
     * @ORM\Column(type="text", nullable=true)
     * @Assert\NotNull
     */
    private $otherCustomName;
[...]
```

For a standard attribute, the `@Assert\NotNull` annotation is not added.
but that's the case for all other custom entities.

**Comment by lologhi** [comment]
Same problem for my custom fields:

```
    Thing:
        properties:
            wikidataId: { range: Text, nullable: true }
```

will generate:
```
    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $wikidataId;
```

I was able to change that by temporarily tweaking the `TypesGenerator.php`.

Original one needs a specific `$cardinality` to set `$isNullable` to true. But with a simple Text field, I don't have cardinality.
```
            if (isset($propertyConfig['nullable']) && false === $propertyConfig['nullable']) {
                $isNullable = false;
            } else {
                $isNullable = !in_array($cardinality, [
                    CardinalitiesExtractor::CARDINALITY_1_1,
                    CardinalitiesExtractor::CARDINALITY_1_N,
                ], true);
            }
```

So I just set that to: 
```
            if (isset($propertyConfig['nullable']) && false === $propertyConfig['nullable']) {
                $isNullable = false;
            } else {
                $isNullable = true;
            }
```

**Which is a bad idea** but for my simple schema did the job.

**Comment by JulienCheny** [comment]
Thank you for the answer.

**Comment by dunglas** [comment]
Closed by #160
